File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    yearning_hash = {}
    answer = []
    for k, v in zip(name, yearning):
        yearning_hash[k] = v
    
    for p in photo:
        score = 0
        for i in p:
            if i in yearning_hash:
                score += yearning_hash[i]
        answer.append(score)
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.swift
import Foundation

func solution(_ name:[String], _ yearning:[Int], _ photo:[[String]]) -> [Int] {
    var yearningDict = Dictionary(uniqueKeysWithValues: zip(name, yearning))
    return photo.map{ $0.map{ yearningDict[$0, default: 0] }.reduce(0, +) }
}